### ISSUE TITLE
Feature Request: support {{if !myvariable}} 

### ISSUE THREAD

**Issue by schlichtanders** [tlienart/JuDoc.jl#899]
it would be nice to have a simple way to negate an if statement.

Currently you have to write
```
{{if myvariable}} {{else}} ... {{end}} 
```

It would be nice to just write
```
{{if !myvariable}} ... {{end}}
```

My current Franklin.jl just stops working as soon as I try this
```
ERROR: Franklin.HTMLBlockError("I found a lonely {{end}}.")
```
using Franklin v0.10.59 and Julia 1.6.1

**Comment by schlichtanders** [comment]
thank you.

If you would allow for such a `e"""..."""` syntax to make use of everything, why not directly supporting it, without an extra `e"""..."""` wrapping?

It should be the same amount of security and way more intuitive to use.

**Comment by tlienart** [comment]
It's not a question of security, it's a question of parsing. There needs to be a way for Franklin to know where are the boundaries of the function. And there needs to be a distinction with `{{func """hello"""}}` which passes the string "hello" to the function.

**Comment by tlienart** [comment]
@schlichtanders I've now implemented this in Xranklin (the in-dev next version) and it looks like this:

same as before:
```
{{whatever x y z}}  # same as before --> ["x", "y", "z"] passed to `hfun_whatever` to do what it wants with it
{{if x}} # same as before --> check local variable "x" if it evaluates to true etc
```

new:
```
{{if e"$x && $y"}} --> checks get_local_var(:x) && get_local_var(:y), if true run if block
{{whatever x e"foo($z)"}} --> passes ["x", "$(foo(get_local_var(:z)))"] to the hfun_whatever
```

so basically the new stuff are these "e-string" where you can specify some Julia logic and indicate what variables to get from the local environment using a `$`. 

Any feedback on this?

**Comment by schlichtanders** [comment]
@tlienart thank you for pinging - awesome to see that this is coming soon.

one feedback immediately popping into my mind: Not everything is nicely representable as a string.
If my function `foo` for whatever reason likes to return an array or struct, wouldn't it be nicer to just have
```
{{whatever x e"foo($z)"}} --> passes ["x", foo(get_local_var(:z))] to the hfun_whatever
```

**Comment by tlienart** [comment]
Yes I agree & this is actually just a matter of where things are interpreted; what actually gets passed is an array of string one of which is an "e-string", the hfun then does what it wants with it but most likely will just evaluate it and there it will be whatever value structure or whatever.

So I misrepresented this in my previous comment, it will in fact be similar to what you described

Thanks for the feedback!
